slsDetectorSoftware/src/slsDetector.cpp

```diff
@@ -2666,6 +2677,129 @@ uint32_t slsDetector::getADCEnableMask() {
     return shm()->adcEnableMask;
 }
 
+void slsDetector::setADCInvert(uint32_t value) {
+    uint32_t arg = value;
+    FILE_LOG(logDEBUG1) << "Setting ADC Invert to 0x" << std::hex << arg << std::dec;
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_SET_ADC_INVERT, &arg, nullptr);
+    }
+}
+
+uint32_t slsDetector::getADCInvert() {
+    uint32_t retval = -1;
+    FILE_LOG(logDEBUG1) << "Getting ADC Invert";
+    if (shm()->onlineFlag == ONLINE_FLAG) {
+        sendToDetector(F_GET_ADC_INVERT, nullptr, 0, &retval, sizeof(retval));
```

Comment:
Also here no size and no ref on retval, the following line is enough
sendToDetector(F_GET_ADC_INVERT, nullptr, retval);